pantos/client/library/blockchains/factory.py

```diff
@@ -1,35 +1,48 @@
 """Factory for blockchain clients.
 
 """
-import typing
-
+import semantic_version  # type: ignore
 from pantos.common.blockchains.base import Blockchain
 
 from pantos.client.library.blockchains.base import BlockchainClient
+from pantos.client.library.protocol import get_latest_protocol_version
+from pantos.client.library.protocol import is_supported_protocol_version
 
-_blockchain_clients: typing.Dict[Blockchain, BlockchainClient] = {}
+_blockchain_clients: dict[tuple[Blockchain, semantic_version.Version],
+                          BlockchainClient] = {}
 """Blockchain-specific client objects."""
 
 _blockchain_client_classes = BlockchainClient.find_subclasses()
 """Blockchain-specific client classes."""
 
 
-def get_blockchain_client(blockchain: Blockchain) -> BlockchainClient:
+def get_blockchain_client(
+        blockchain: Blockchain,
+        protocol_version: semantic_version.Version | None = None) \
+        -> BlockchainClient:
     """Factory for blockchain-specific client objects.
 
     Parameters
     ----------
     blockchain : Blockchain
         The blockchain to get the client instance for.
+    protocol_version : semantic_version.Version, optional
```

Comment:
Is there at any point in time the chance that there are multiple versions which are supported? In the current architecture that would mean having at least 2 smart contracts on the same chain.

Reply:
Yes, that will certainly happen after the mainnet launch. The protocol version in testnet will be newer.